Lib/test/test_tools/test_pathfix.py

```diff
@@ -0,0 +1,34 @@
+import unittest, os, subprocess
+from test.test_tools import import_tool, scriptsdir
+import sys
+from test import support
+
+
+class TestPathfixFunctional(unittest.TestCase):
+    script = os.path.join(scriptsdir, 'pathfix.py')
+
+    @classmethod
+    def setupUpClass(cls):
+        self.addCleanup(support.unlink, support.TESTFN)
+
+    def setUp(self):
+        self.temp_file = support.TESTFN
+
+        with open(self.temp_file, 'w') as f:
+            f.write('#! /usr/bin/env python -R\n' + 'print("Hello world")\n')
```

Comment:
I would prefer to have an helper method which takes two arguments: (shebang, pathfix_flags) and return the new shebang. To get tests which look like:

```
def test_keep_flags(self):
  shebang = '#! /usr/bin/env python -R'
  self.assertEqual(
    self.pathfix(shebang, ['-i', '/usr/bin/python']),
    '#! /usr/bin/python')
  self.assertEqual(
    self.pathfix(shebang, ['-i', '/usr/bin/python', '-k']),
    '#! /usr/bin/python -R')
```